lib/ts/recipe/emailverification/api/implementation.ts

```diff
@@ -43,13 +47,12 @@ export default function getAPIInterface(): APIInterface {
                 throw new Error("Session is undefined. Should not come here.");
             }
 
-            let userId = session.getUserId();
-
-            let email = await options.config.getEmailForUserId(userId, userContext);
+            await session.fetchAndSetClaim(EmailVerifiedClaim, userContext);
+            const isVerified = await session.getClaimValue(EmailVerifiedClaim, userContext);
 
             return {
                 status: "OK",
-                isVerified: await options.recipeImplementation.isEmailVerified({ userId, email, userContext }),
+                isVerified: isVerified === true,
```

Comment:
Why not just `isVerified`?

Reply:
`isVerified` may be undefined from a types point of view.

Reply:
I see. So if it is undefined, then would it be right to send the user to the email verification screen? What would happen there?

And if it should never be undefined, maybe we should have a check for that and throw an error in case it is.

Reply:
It can't be undefined, but because of typing complexities, it shows up as a maybe type; I'll add the check/throw.